ql/src/java/org/apache/hadoop/hive/ql/exec/repl/incremental/IncrementalLoadTasksBuilder.java

```diff
@@ -177,10 +174,20 @@ public IncrementalLoadTasksBuilder(String dbName, String tableName, String loadP
       this.log.debug("Added {}:{} as a precursor of barrier task {}:{}",
               taskChainTail.getClass(), taskChainTail.getId(),
               barrierTask.getClass(), barrierTask.getId());
+      // This is making it dependent on the fact that incremental load as to happen before hdfs sync
+      // for external tables and only when everything w.r.t to incremental is done should
+      // hdfs replication start
+      if (loadWork.getPathsToCopyIterator().hasNext()) {
+        taskChainTail.addDependentTask(TaskFactory.get(loadWork, conf));
```

Comment:
This comment doesn't make sense as the caller always try to include externalCopyTasks even if incremental load have some event tasks pending. Also, caller will take care of appending ReplLoadTask if either event or externalCopyTasks are there. So, this comment and code can be removed.

Reply:
done